plugins/modules/dcnm_service_policy.py

```diff
@@ -0,0 +1,1962 @@
+                pol_list, sp_policy_spec
+            )
+            if invalid_params:
+                mesg = "Invalid parameters in playbook: {}".format(
+                    "while processing policy under Service Policy - "
+                    + cfg[0]["name"]
+                    + ", "
+                    + "\n".join(invalid_params)
+                )
+                self.module.fail_json(msg=mesg)
+            pol_list.remove(item["policy"])
+            item["policy"] = policy_info
+
+    def dcnm_sp_validate_input(self):
+
+        """
+        Routine to validate playbook input based on the state. Since each state has a different i
+        config structure, this routine handles the validation based on the given state
+
+        Parameters:
+            None
+
+        Returns:
+            None
+        """
+
+        if None is self.config:
+            return
+
+        cfg = []
+        for item in self.config:
+
+            citem = copy.deepcopy(item)
+
+            cfg.append(citem)
+
+            if self.module.params["state"] == "deleted":
+                if (
+                    item.get("name")
+                    or item.get("node_name")
+                    or item.get("rp_name")
+                ):
+                    # config for delete state is different. So validate deleted state differently
+                    self.dcnm_sp_validate_delete_state_input(cfg)
+            elif self.module.params["state"] == "query":
+                # config for query state is different. So validate query state differently
+                self.dcnm_sp_validate_query_state_input(cfg)
+            # For 'overridden' state, we can have full config for a peering or just service node name alone.
+            # In the formar case go down to 'else' block to validate the full config
+            elif (self.module.params["state"] == "overridden") and (
+                item.get("name", None) is None
+            ):
+                pass
+            else:
+                self.dcnm_sp_validate_sp_input(cfg)
+            cfg.remove(citem)
+
+    def dcnm_sp_validate_sp_input(self, cfg):
+
+        """
+        Routine to validate the playbook input. This routine updates self.sp_info
+        with validated playbook information by defaulting values not included
+
+        Parameters:
+            cfg (dict): The config from playbook
+
+        Returns:
+            None
+        """
+
+        sp_spec = dict(
+            name=dict(required=True, type="str"),
+            node_name=dict(required=True, type="str"),
+            rp_name=dict(required=True, type="str"),
+            src_vrf=dict(required=True, type="str"),
+            dest_vrf=dict(required=True, type="str"),
+            src_network=dict(required=True, type="str"),
+            dest_network=dict(required=True, type="str"),
+            next_hop=dict(required=True, type="ipv4"),
+            rev_next_hop=dict(required=True, type="ipv4"),
```

Comment:
Looks like `rev_next_hop` is not listed as required in the documentation.  Please double check all required parameters in the code to make sure the docs match.

Reply:
Done